packages/plugin-ext/src/hosted/node/hosted-plugin-process.ts

```diff
@@ -113,10 +113,7 @@ export class HostedPluginProcess implements ServerPluginRunner {
     private fork(options: IPCConnectionOptions): cp.ChildProcess {
 
         // create env and add PATH to it so any executable from root process is available
-        const env = createIpcEnv();
-        env.PATH = process.env.PATH;
-        // add HOME to env since some plug-ins need to read files from user's home dir
-        env.HOME = process.env.HOME;
+        const env = createIpcEnv({ env: process.env });
```

Comment:
If I need `GOPATH` and `GOROOT` should I copy them separately?

Reply:
I think there is inception mechanism when copying all variables (like `HOSTED_PLUGIN` being added in new hosted process, so it will spawn another process, etc) and some other variables shouldn't be visible too

so either we need an inclusion filter (like today) or an exclusion filter (so you can inherit and find `GOPATH` or `GOROOT` but we can't copy them directly

Reply:
ok. I will do in the same way as we did with `PATH` and `HOME` variables. 
```
        const env = createIpcEnv();

        env.PATH = process.env.PATH;
        // add HOME to env since some plug-ins need to read files from user's home dir
        env.HOME = process.env.HOME;

        env.GOPATH = process.env.GOPATH;
        env.GOROOT = process.env.GOROOT;
```

Reply:
I think it'll be better to use exclude filters now instead of include filters.

Reply:
In case of exclusion which environment variables are not supposed to be passed in the new process?

Reply:
The one with Plugins (like `HOSTED_PLUGIN`, `THEIA_PLUGINS`, `THEIA_DEFAULT_PLUGINS`)
maybe some other I don't have the full list in mind.
probably as well the one with the port to set